### ISSUE TITLE
FastifyAdapter type error

### ISSUE THREAD

**Issue by igor17400** [nestjs/nest#10596]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Following this [tutorial](https://medium.com/the-nobodys-of-tech/nestjs-fastify-aws-lambda-86e5c460cf57) to implement nestJS + fastify as lambda function is not working due to FastifyAdapter type error.

It seems that this bug has already been reported in https://github.com/nestjs/nest/issues/3671 and in https://github.com/nestjs/nest/issues/3655, however, no solution has been provided.

### Minimum reproduction code

https://gist.github.com/igor17400/c070c7808f4a61911c08dc2d77f08f4e

### Expected behavior

No error when declaring an instance on `FastifyAdapter()` constructor. 

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [X] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### Packages versions

```

{
  "name": "forest",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {},
  "private": true,
  "devDependencies": {
    "@nestjs/schematics": "^9.0.0",
    "@nestjs/testing": "^9.0.0",
    "@nrwl/cli": "15.2.1",
    "@nrwl/cypress": "15.2.1",
    "@nrwl/eslint-plugin-nx": "15.2.1",
    "@nrwl/jest": "15.2.1",
    "@nrwl/js": "15.2.1",
    "@nrwl/linter": "15.2.1",
    "@nrwl/nest": "15.2.1",
    "@nrwl/next": "^15.2.1",
    "@nrwl/node": "15.2.1",
    "@nrwl/react": "15.2.1",
    "@nrwl/web": "15.2.1",
    "@nrwl/workspace": "15.2.1",
    "@testing-library/react": "13.4.0",
    "@types/jest": "28.1.1",
    "@types/node": "18.7.1",
    "@types/react": "18.0.20",
    "@types/react-dom": "18.0.6",
    "@typescript-eslint/eslint-plugin": "^5.36.1",
    "@typescript-eslint/parser": "^5.36.1",
    "aws-lambda": "^1.0.7",
    "babel-jest": "28.1.1",
    "cypress": "^11.0.0",
    "eslint": "~8.15.0",
    "eslint-config-next": "13.0.0",
    "eslint-config-prettier": "8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jsx-a11y": "6.6.1",
    "eslint-plugin-react": "7.31.8",
    "eslint-plugin-react-hooks": "4.6.0",
    "jest": "28.1.1",
    "jest-environment-jsdom": "28.1.1",
    "nx": "15.2.1",
    "prettier": "^2.6.2",
    "react-test-renderer": "18.2.0",
    "ts-jest": "28.0.5",
    "ts-node": "10.9.1",
    "typescript": "~4.8.2"
  },
  "dependencies": {
    "@nestjs/common": "^9.0.0",
    "@nestjs/core": "^9.0.0",
    "@nestjs/platform-express": "^9.0.0",
    "@nestjs/platform-fastify": "^9.2.0",
    "@tailwindcss/typography": "^0.5.8",
    "autoprefixer": "^10.4.13",
    "aws-lambda-fastify": "^2.2.0",
    "aws-serverless-fastify": "^3.0.0",
    "core-js": "^3.6.5",
    "fastify": "^4.10.2",
    "next": "13.0.0",
    "postcss": "^8.4.19",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "reflect-metadata": "^0.1.13",
    "regenerator-runtime": "0.13.7",
    "rxjs": "^7.0.0",
    "tailwindcss": "^3.2.4",
    "tslib": "^2.3.0"
  }
}


```

### Node.js version

v18.11.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
show us the error, please.

**Comment by micalevisk** [comment]
this issue will be fixed after #10583 I just tested it.

**Comment by kamilmysliwiec** [comment]
https://github.com/nestjs/nest/pull/10583
